### ISSUE TITLE
GPU_BlitRectX Incorrect Flip

### ISSUE THREAD

**Issue by DiegoAce** [grimfang4/sdl-gpu#56]
I believe the recent commit fixing the pivot point messed up the result of using GPU_FlipEnum in GPU_BlitRectX.
I think the code that adjusts for the flip in GPU_BlitRectX should be changed to something like this:
```
if(flip_direction & GPU_FLIP_HORIZONTAL)
{
    scale_x = -scale_x;
    dx += dw;
    pivot_x = w - pivot_x;
}
if(flip_direction & GPU_FLIP_VERTICAL)
{
    scale_y = -scale_y;
    dy += dh;
    pivot_y = h - pivot_y;
}
```
This behaves like using SDL_RendererFlip in SDL_RenderCopyEx.

**Comment by grimfang4** [comment]
Yes, that seems appropriate.  Can you make a pull request with this change?  My time to do testing is limited, so I'd like to streamline that.

**Comment by DiegoAce** [comment]
Sure thing!

**Comment by grimfang4** [comment]
Merged pull request in e1fea9003831092a71ac57c7ae03066a0883d5bb
